### PR TITLE
Add content database in RPM and DEB packages

### PR DESCRIPTION
**Pull request by diegodfrf**
|Related issue|
|---|
|[#20747](https://github.com/wazuh/wazuh/issues/20747)|

## Description

Adds the option to download the vulnerability detector content, compress it into a tar.xz file, add it to the installer, and unzip it at installation.

## Tests

* New flag --download-content in DEB

image

* New flag  --download-content in RPM

image

* Generation DEB package successful

image

* Installation from DEB packages successful

image

image

* Generation RPM package successful

image

* Installation from RPM packages successful

image

image

### CODE REVIEW

File: debs/SPECS/wazuh-manager/debian/postinst [hunk 1]
```diff
@@ -15,6 +15,7 @@ case "$1" in
     OSMYSHELL="/sbin/nologin"
     SCRIPTS_DIR="${WAZUH_GLOBAL_TMP_DIR}/manager_installation_scripts"
     SCA_BASE_DIR="${SCRIPTS_DIR}/sca"
+    VD_FILENAME='vd_1.0.0_vd_4.8.0.tar.xz'
```
Comment (pereyra-m) [inline debs/SPECS/wazuh-manager/debian/postinst:18]: We should find the best way to manager the file versioning.

File: debs/SPECS/wazuh-manager/debian/postinst [hunk 2]
```diff
@@ -71,6 +72,13 @@ case "$1" in
         chown ${USER}:${GROUP} ${DIR}/queue/db/global.db*
     fi
 
+    if [ -f "${DIR}/${VD_FILENAME}" ]; then
+        tar -xf ${DIR}/${VD_FILENAME} -C ${DIR}
+        chown ${USER}:${GROUP} ${DIR}/queue/vd
+        chown ${USER}:${GROUP} ${DIR}/queue/vd-updater
```
Comment (pereyra-m) [inline debs/SPECS/wazuh-manager/debian/postinst:78]: This folder isn't required.

File: rpms/SPECS/wazuh-manager.spec [hunk 3]
```diff
@@ -298,6 +299,14 @@ if [ $1 = 2 ]; then
   fi
 fi
 
+VD_FILENAME='vd_1.0.0_vd_4.8.0.tar.xz'
```
Comment (pereyra-m) [inline rpms/SPECS/wazuh-manager.spec:302]: We should find the best way to manager the file versioning.

File: rpms/SPECS/wazuh-manager.spec [hunk 4]
```diff
@@ -298,6 +299,14 @@ if [ $1 = 2 ]; then
...
+if [ -f "%{_localstatedir}/${VD_FILENAME}" ]; then
+    tar -xf %{_localstatedir}/${VD_FILENAME} -C %{_localstatedir}
+    chown wazuh:wazuh %{_localstatedir}/queue/vd
+    chown wazuh:wazuh %{_localstatedir}/queue/vd-updater
```
Comment (pereyra-m) [inline rpms/SPECS/wazuh-manager.spec:306]: This folder isn't required.

**Review by pereyra-m — COMMENTED** [review]
Gj!

File: debs/SPECS/wazuh-manager/debian/postinst (hunk 1 above)
Comment (diegodfrf) [reply debs/SPECS/wazuh-manager/debian/postinst:18]: For the moment it will be handled like this.

File: rpms/SPECS/wazuh-manager.spec (hunk 3 above)
Comment (diegodfrf) [reply rpms/SPECS/wazuh-manager.spec:302]: For the moment it will be handled like this.

File: rpms/SPECS/wazuh-manager.spec (hunk 4 above)
Comment (diegodfrf) [reply rpms/SPECS/wazuh-manager.spec:306]: As per the requirements, the vd_updater folder needs to be compressed.

File: debs/SPECS/wazuh-manager/debian/postinst (hunk 2 above)
Comment (diegodfrf) [reply debs/SPECS/wazuh-manager/debian/postinst:78]: According to the requirements, the vd_updater folder must be compressed.

File: debs/SPECS/wazuh-manager/debian/postinst (hunk 1 above)
Comment (pereyra-m) [reply debs/SPECS/wazuh-manager/debian/postinst:18]: Ok, thank you

File: debs/SPECS/wazuh-manager/debian/postinst (hunk 2 above)
Comment (pereyra-m) [reply debs/SPECS/wazuh-manager/debian/postinst]: Ok, thank you.

File: rpms/SPECS/wazuh-manager.spec (hunk 3 above)
Comment (pereyra-m) [reply rpms/SPECS/wazuh-manager.spec:309]: Ok, thank you.

File: rpms/SPECS/wazuh-manager.spec (hunk 4 above)
Comment (pereyra-m) [reply rpms/SPECS/wazuh-manager.spec]: Ok, thank you.

**Review by pereyra-m — APPROVED** [review]
LGTM!

**Comment by pereyra-m** [comment]
## Testing :green_circle: 

Packages generated by @MiguelazoDS .

The `.deb` package was installed and the content was present

```
root@ubuntu-jammy:/var/ossec# du -hs queue/vd*
4.5G	queue/vd
116K	queue/vd_updater
root@ubuntu-jammy:/var/ossec# du -hs queue/vd/*
1.5G	queue/vd/candidates
2.9G	queue/vd/cves
131M	queue/vd/descriptions
48K	queue/vd/inventory
34M	queue/vd/remediations
56K	queue/vd/translations
```

Also, the `content_module` reads the offset properly




The `.rpm` package was installed and the content was present

```
[root@localhost ossec]# du -hs queue/vd*
4.5G	queue/vd
4.2M	queue/vd_updater
[root@localhost ossec]# du -hs queue/vd/*
1.5G	queue/vd/candidates
2.9G	queue/vd/cves
135M	queue/vd/descriptions
4.1M	queue/vd/inventory
38M	queue/vd/remediations
4.1M	queue/vd/translations
```

**Pull request merged**